### DISCUSSION TITLE
How Can I get clip volume  of rang3d from active view correctly?

### DISCUSSION THREAD

**Discussion by NealCpf** [iTwin/itwinjs-core#1481] [Q&A] [answered]
hi , I used this way below to get active view clip volume which I put it into . 

**let clipVolumRange: Range3d | undefined = vp?.view.details.clipVector?.boundingRange;**
            **console.log(clipVolumRange?.center);
            console.log(clipVolumRange?.low);
            console.log(clipVolumRange?.high);**

But the range3d always returned the value below ,whatever I put the clipvolume to anywhere.
**Point3d {x: 0, y: 0, z: 0}
CabelTableTool.ts:40 Point3d {x: 1e+200, y: 1e+200, z: 1e+200}
CabelTableTool.ts:41 Point3d {x: -1e+200, y: -1e+200, z: -1e+200}
CabelTableTool.ts:43 1**

**Comment by pmconne** [comment, marked as answer]
That's a null range. The bounding range property is only used in very specific circumstances. What are you trying to accomplish?

**Reply by NealCpf** [reply]
Hi, thanks for your reply.  I put this clip volume with view clip tool from keyin command. But I want to get the elements in the clip volume.
So I need the clip volume range3d to query the element with ecsql.  Do you have any other idea to get my goal?

![image](https://user-images.githubusercontent.com/30515187/119312229-a8f1a800-bca4-11eb-9dc4-fef86df86133.png)

**Reply by NealCpf** [reply]
Thanks you again! I've resolved this issue with under code. 

**const range = vp?.computeViewRange();
        const clip = vp?.view.getViewClip();
        if (undefined !== clip) {
            const clipRange = ClipUtilities.rangeOfClipperIntersectionWithRange(clip, range!);
            if (!clipRange.isNull)
                range!.setFrom(clipRange);
        }**
